### ISSUE TITLE
Upgrade Akka framework to 2.3.x

### ISSUE THREAD

**Issue by gvagenas** [RestComm/Restcomm-Connect#1198]

**Comment by muhammadbilal19** [comment]
@gvagenas @deruelle not sure how urgent you guys needs this task to be done, but I am currently in a process of learning  akka and other Lightbend products and i would like to contribute to this task along with #2274. Although i am pretty new to akka framework but i guess this is a really good chance for me to learn akka and along with RestComm implementation of akka as well.I might need some help to understand current implementation so i will disturb you and @gvagenas lot of time during this task :)

**Comment by gvagenas** [comment]
@muhammadbilal19 As far as I know latest Akka needs JDK8 and we cannot move to JDK8 right now. Can you check if they provide a version compiled for JDK7?

**Comment by muhammadbilal19** [comment]
@gvagenas yes you are right since akka 2.4 jdk8 is supported. OK i will wait for #2269 to complete.
